deps/rabbitmq_auth_backend_ldap/src/rabbit_auth_backend_ldap_mgmt.erl

```diff
@@ -0,0 +1,196 @@
+            end,
+            TlsOpts2 = case CaCertPemData of
+                undefined ->
+                    TlsOpts1;
+                CaCertPems when is_list(CaCertPems) ->
+                    F0 = fun (P) ->
+                        case public_key:pem_decode(P) of
+                            [{'Certificate', CaCertDerEncoded, not_encrypted}] ->
+                                {true, CaCertDerEncoded};
+                            _Unexpected ->
+                                throw({bad_request, "unexpected cacert_pem_data passed to "
+                                                    "/ldap/validate/simple-bind ssl_options.cacerts"})
+                        end
+                    end,
+                    CaCertsDerEncoded = lists:filtermap(F0, CaCertPems),
+                    [{cacerts, CaCertsDerEncoded} | TlsOpts1];
+                _ ->
+                    TlsOpts1
+            end,
+            TlsOpts3 = case maps:get(<<"verify">>, SslOptionsMap, undefined) of
+                undefined ->
+                    TlsOpts2;
+                Verify ->
+                    VerifyStr = unicode:characters_to_list(Verify),
+                    [{verify, list_to_existing_atom(VerifyStr)} | TlsOpts2]
+            end,
+            TlsOpts4 = case maps:get(<<"server_name_indication">>, SslOptionsMap, disable) of
+                disable ->
+                    TlsOpts3;
+                SniValue ->
+                    SniStr = unicode:characters_to_list(SniValue),
+                    [{server_name_indication, SniStr} | TlsOpts3]
+            end,
+            TlsOpts5 = case maps:get(<<"depth">>, SslOptionsMap, undefined) of
+                undefined ->
+                    TlsOpts4;
+                DepthValue ->
+                    Depth = rabbit_data_coercion:to_integer(DepthValue),
+                    [{depth, Depth} | TlsOpts4]
+            end,
+            TlsOpts6 = case maps:get(<<"versions">>, SslOptionsMap, undefined) of
+                undefined ->
+                    TlsOpts5;
+                VersionStrs when is_list(VersionStrs) ->
+                    F1 = fun (VStr) ->
+                        try
+                            {true, list_to_existing_atom(VStr)}
+                        catch error:badarg ->
+                            throw({bad_request, "invalid TLS version passed to "
+                                                "/ldap/validate/simple-bind ssl_options.versions"})
+                        end
+                    end,
+                    Versions = lists:filtermap(F1, VersionStrs),
+                    [{versions, Versions} | TlsOpts5]
+            end,
+            TlsOpts7 = case maps:get(<<"ssl_hostname_verification">>, SslOptionsMap, undefined) of
+                undefined ->
+                    TlsOpts6;
+                "wildcard" ->
+                    [{customize_hostname_check, [{match_fun, public_key:pkix_verify_hostname_match_fun(https)}]} | TlsOpts6]
```

Comment:
It might be nice to have an extra branch here which throws bad-request with a nice error message, so that if you set `ssl_hostname_verification` to something that isn't `"wildcard"` you don't get the case-clause error instead. (Also the `end` below could be unindented one level)